addons/account_edi_peppol/models/account_edi_format.py

```diff
@@ -0,0 +1,266 @@
+            'res_id': invoice._origin.id,
+            'mimetype': 'application/xml'
+        })}
+
+    # -------------------------------------------------------------------------
+    # Import
+    # -------------------------------------------------------------------------
+
+    def _is_peppol(self, filename, tree):
+        tag = tree.tag == '{urn:oasis:names:specification:ubl:schema:xsd:Invoice-2}Invoice'
+        profile = tree.xpath("//*[local-name()='ProfileID']")
+        profile = profile[0].text == 'urn:fdc:peppol.eu:2017:poacc:billing:01:1.0' if profile else False
+
+        return self.code == 'peppol_3_10' and tag and profile
+
+    def _decode_peppol(self, tree, invoice):
+
+        def _get_ubl_namespaces():
+            ''' If the namespace is declared with xmlns='...', the namespaces map contains the 'None' key that causes an
+            TypeError: empty namespace prefix is not supported in XPath
+            Then, we need to remap arbitrarily this key.
+
+            :param tree: An instance of etree.
+            :return: The namespaces map without 'None' key.
+            '''
+            namespaces = tree.nsmap
+            namespaces['inv'] = namespaces.pop(None)
+            return namespaces
+
+        self.ensure_one()
+        namespaces = _get_ubl_namespaces()
+
+        type = 'in_invoice'
+
+        default_journal = invoice.with_context(default_move_type=type)._get_default_journal()
+
+        with Form(invoice.with_context(default_move_type=type, default_journal_id=default_journal.id)) as invoice_form:
+            # Reference
+            elements = tree.xpath('//cbc:ID', namespaces=namespaces)
+            if elements:
+                invoice_form.ref = elements[0].text
+
+            # Dates
+            elements = tree.xpath('//cbc:IssueDate', namespaces=namespaces)
+            if elements:
+                invoice_form.invoice_date = elements[0].text
+            elements = tree.xpath('//cbc:DueDate', namespaces=namespaces)
+            if elements:
+                invoice_form.invoice_date_due = elements[0].text
+
+            # Currency
+            elements = tree.xpath('//cbc:DocumentCurrencyCode', namespaces=namespaces)
+            currency_code = elements and elements[0].text or ''
+            currency = self.env['res.currency'].search([('name', '=', currency_code.upper())], limit=1)
+            if currency:
+                invoice_form.currency_id = currency
+
+            # Partner
+            partner_element = tree.xpath('//cac:AccountingSupplierParty/cac:Party', namespaces=namespaces)
+            if partner_element:
+                domains = []
+                partner_element = partner_element[0]
+                elements = partner_element.xpath('//cac:AccountingSupplierParty/cac:Party//cbc:Name', namespaces=namespaces)
+                if elements:
+                    partner_name = elements[0].text
+                    domains.append([('name', 'ilike', partner_name)])
```

Comment:
time to add some helpers in `account_edi` I think...
`_retrieve_tax`, `_retrieve_partner`, `_retrieve_product` ...

Reply:
But if they need to be overriden, then what's the use of a helper ?

Reply:
Want to have some helpers to be used in any edi

Reply:
Yes, but this is an import function, so it would be dependent on the format how to retrieve the taxes. The helper would need to be a hook, but I don't see how it would be useful here unless we make a fully generic import with only hooks to override to retrieve taxes, partners, products, etc but then it's out of the scope of this task I think.

Reply:
Here is my suggestion:
```
def _retrieve_partner(self, name=None, phone=None, mail=None, vat=None):
    domains = []
    for value, domain in (
            (name, ('name', 'ilike', name)),
            (phone, ('phone', '=', phone), ('mobile', '=', phone)),
            (mail, ('email', '=', mail)),
            (vat, ('vat', 'like', vat)),
    ):
        if value is not None:
            domains.append(domain)

    if domains:
        return self.env['res.partner'].search(expression.OR(domains), limit=1)
```
  This method could be used in any import

Reply:
Then, it will simplify your code:
```
partner = self._retrieve_partner(
    name=_find_value('//cac:AccountingSupplierParty/cac:Party//cbc:Name'),
    phone=_find_value('//cac:AccountingSupplierParty/cac:Party//cbc:Telephone'),
    mail=_find_value('//cac:AccountingSupplierParty/cac:Party//cbc:ElectronicMail'),
    vat=_find_value('//cac:AccountingSupplierParty/cac:Party//cbc:ID'),
)
```

Reply:
Same for taxes/product, what do you think?